pkg/backporter/backporter.go

```diff
@@ -553,30 +558,118 @@ func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String, m
 			handleError(w, err, fmt.Sprintf("unable to fetch bug details- Bug#%d", bugID), http.StatusNotFound, endpoint, bugID, m)
 			return
 		}
-		// Create a clone of the bug
-		cloneID, err := client.CloneBug(bug)
+		if !allTargetVersions.Has(req.FormValue("release")) {
+			handleError(w, err, fmt.Sprintf("invalid argument - not a valid TargetRelease - %s", req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+
+		// Get clones and sort them
+		clones, err := client.GetAllClones(bug)
 		if err != nil {
-			handleError(w, err, "clone creation failed", http.StatusInternalServerError, endpoint, bugID, m)
+			handleError(w, err, fmt.Sprintf("unable to convert \"ID\" parameter from string to int: %s", req.FormValue("ID")), http.StatusBadRequest, req.URL.Path, 0, m)
 			return
 		}
-		targetRelease := bugzilla.BugUpdate{
-			TargetRelease: []string{
-				req.FormValue("release"),
-			},
+		sort.SliceStable(clones, func(i, j int) bool {
+			if clones[i].TargetRelease == nil && clones[j].TargetRelease == nil {
+				return false
+			} else if clones[i].TargetRelease == nil {
+				return true
+			} else if clones[j].TargetRelease == nil {
+				return false
+			}
+			return clones[i].TargetRelease[0] < clones[j].TargetRelease[0]
+		})
+		toCloneMajRelease, err := getMajorRelease(req.FormValue("release"))
```

Comment:
There is no penality for longer variable names so please make sure they are clear, here `toCloneMajorRelease` (although as commented elsewhere that is wrong anyways, because its a majorMinor release, majorRelease would just be the `4`)